hw2/01-coins.js

```diff
@@ -1,7 +1,131 @@
-/** Exercise 01 - Coins **/
+/* eslint-disable no-console */
+/** Exercise 01 - Coins * */
 
+/**
+ * Removes as many dollars as possible and returns
+ * amount removed and amount remaing
+ */
+const removeDollar = (num) => {
+  let dollars = 0;
+  let rem = num;
+  while (rem > 100) {
+    rem -= 100;
+    dollars += 1;
+  }
+  return [dollars, rem];
+};
+
+/**
+ * Removes as many quarters as possible and returns
+ * amount removed and amount remaing
+ */
+const removeQuart = (num) => {
+  let quarters = 0;
+  let rem = num;
+  while (rem > 25) {
+    rem -= 25;
+    quarters += 1;
+  }
+  return [quarters, rem];
+};
+
+/**
+ * Removes as many dimes as possible and returns
+ * amount removed and amount remaing
+ */
+const removeDime = (num) => {
+  let dimes = 0;
+  let rem = num;
+  while (rem > 10) {
+    rem -= 10;
+    dimes += 1;
+  }
+  return [dimes, rem];
+};
+
+/**
+ * Removes as manu nickels as possible and returns
+ * amount removed and amount remaing
+ */
+const removeNickel = (num) => {
+  let nickels = 0;
+  let rem = num;
+  while (rem > 5) {
+    rem -= 5;
+    nickels += 1;
+  }
+  return [nickels, rem];
+};
+
+/**
+ * Removes as many pennies as possible and returns
+ * amount removed and amount remaing
+ */
+const removePenny = (num) => {
+  let pennies = 0;
+  let rem = num;
+  while (rem >= 1) {
+    rem -= 1;
+    pennies += 1;
+  }
+  return pennies;
+};
+
+// Formats the string according to amount of change
+const formatChange = (dollars, quarters, dimes, nickels, pennies) => {
```

Comment:
Great way to go about handling the plurals. I did something similar but for each conditional, the next conditional would check all the currency types below it, just for the string formatting to be accurate with their commas and spaces. I didn't think about it initially but I had to break it down into each use cases, for example: 1 quarter, 1 dime, nothing else, I didn't want my string to show up like "1 quarter, 1 dime, ". Its not a big deal in my opinion but I get so wrapped up into certain use cases sometimes that I will spend the extra time trying to figure out the best way to format the strings lol.